subsys/bluetooth/host/id.c

```diff
@@ -701,6 +701,74 @@ void bt_id_pending_keys_update(void)
 	}
 }
 
+static int hci_id_del(const bt_addr_le_t *addr)
+{
+	struct bt_hci_cp_le_rem_dev_from_rl *cp;
+	struct net_buf *buf;
+
+	BT_DBG("addr %s", bt_addr_le_str(addr));
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_REM_DEV_FROM_RL, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	bt_addr_le_copy(&cp->peer_id_addr, addr);
+
+	return bt_hci_cmd_send_sync(BT_HCI_OP_LE_REM_DEV_FROM_RL, buf, NULL);
+}
+
+static int hci_id_overwrite(struct bt_keys *keys)
+{
+	int err = hci_id_add(keys->id, &keys->addr, keys->irk.val);
+
+	switch (err) {
+		case 0: {
+			/* Success */
+			return 0;
+		}
+
+		case -EIO: {
+			/* Delete the identity in the controller's resolve list since it already
+			 * exists to make sure that a potentially new IRK is stored.
+			 */
+			err = hci_id_del(&keys->addr);
+
+			switch (err) {
+				case 0: {
+					/* Success */
+					return 0;
+				}
+
+				default: {
+					BT_ERR("Failed to remove an existing IRK in the contr.");
+					return err;
+				}
+			}
+
+			err = hci_id_add(keys->id, &keys->addr, keys->irk.val);
+
+			switch (err) {
+				case 0: {
+					/* Success */
+					return 0;
+				}
+
+				default: {
+					BT_ERR("Failed to overwrite an existing IRK in the contr.");
+					return err;
+				}
+			}
+		}
+
+		default: {
+			BT_ERR("Failed to add IRK to controller");
+			return err;
+		}
+	}
```

Comment:
It isn't explicitely stated, but I think that curly brackets in these cases can be removed to better match the Linux Coding Style that Zephyr also follows:
https://www.kernel.org/doc/html/v4.10/process/coding-style.html#placing-braces-and-spaces

Reply:
Ok. Will do.